### ISSUE TITLE
Rework windows build/linkage

### ISSUE THREAD

**Issue by kratman** [pybamm-team/pybammsolvers#18]
The windows wheel build and delivery should be reworked to remove the usage of VCPKG.

Reference issue: https://github.com/pybamm-team/PyBaMM/issues/3603

**Comment by agriyakhetarpal** [comment]
The first part here is to use just one Vcpkg registry instead of two, but if we can remove both at the same time, so be it.

Switching to Meson might have been great here, which would allow us to use git submodules via Meson subprojects to do it.

**Comment by agriyakhetarpal** [comment]
While we are at it, another thing to do with this is to switch to Clang instead of MSVC

**Comment by kratman** [comment]
Yeah I am re-working a bunch of the build stuff. I will take care of this one soon
